include/eve/platform.hpp

```diff
@@ -12,28 +12,43 @@
 
 namespace eve
 {
+  //================================================================================================
+  //! @addtogroup arch
+  //! @{
+  //! @struct platform
+  //! @brief  Platform specific constexpr information
+  //!
+  //! Provides constexpr information about the status (supported or not) of various platform
+  //! computing capability.
+  //! @}
+  //================================================================================================
   struct platform
   {
+    //! @brief Evaluates to `true` if current platform supports NaN values
 #if defined(EVE_NO_NANS)
```

Comment:
Is this rly a thing? Where?

Reply:
A32 ARM doesn't support denormal values in SIMD register  for example. ARM code and tests use this supports_denormals cause if not, they fail.

The others are also meant to be passed by users in order to globally control the fact our SIMD functions care or do not care of some corner cases in FP to win some speed globally. 

I should add doc for the macro or if not this struct looks strange.